modules/electromagnetics/include/bcs/VectorTransientAbsorbingBC.h

```diff
@@ -0,0 +1,55 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "VectorIntegratedBC.h"
+#include <complex>
+
+/**
+ *  First order transient absorbing boundary condition for nonlinear vector variables
+ */
+class VectorTransientAbsorbingBC : public VectorIntegratedBC
+{
+public:
+  static InputParameters validParams();
+
+  VectorTransientAbsorbingBC(const InputParameters & parameters);
+
+protected:
+  virtual Real computeQpResidual() override;
+  virtual Real computeQpJacobian() override;
+  virtual Real computeQpOffDiagJacobian(unsigned int jvar) override;
+
+  /// Intrinsic impedance of the infinite medium (default is the admittance of free space)
+  const Function & _admittance;
+
+  /// Variable field component (real or imaginary)
+  MooseEnum _component;
+
+  /// Coupled field vector variable
+  const VectorVariableValue & _coupled_val;
+  /// Coupled field vector variable id
+  unsigned int _coupled_var_num;
+
+  /// Vector field variable time derivative
+  const VectorVariableValue & _u_dot;
+
+  /// Coupled vector field variable time derivative
+  const VectorVariableValue & _coupled_dot;
+
+  /// Vector field variable dot du
```

Comment:
not sure what that is?
dudt du ?

Reply:
what is Du?

Reply:
It's the derivative with respect to u_j (where u is the solution variable) of the time derivative of u_k (where j and k are the indices of the finite element coefficients of u), and is used in the Jacobian contribution of this object. It's value depends on the time stepping method. See the [Kernel documentation](https://mooseframework.inl.gov/syntax/Kernels/index.html#time-derivative-kernels) for more information. Would you like me to change the comment here to something else?